src/v/cluster/tx_types.h

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+#include "kafka/protocol/types.h"
+#include "model/fundamental.h"
+#include "model/record.h"
+namespace cluster {
```

Comment:
any thoughts on moving to cluster::tx while we are here (like in rm_stm_types)

Reply:
i tried it, it is massive code churn so maybe we can defer that a bit ?